lib/api_account.go

```diff
@@ -26,31 +28,172 @@ func GetAccountHandler(s *sebakstorage.LevelDBBackend) http.HandlerFunc {
 		}
 
 		var err error
-
 		var ba *BlockAccount
-		if ba, err = GetBlockAccount(storage, address); err != nil {
-			http.Error(w, "Error reading request body", http.StatusInternalServerError)
-			return
-		}
-
-		var s []byte
-		if s, err = ba.Serialize(); err != nil {
-			http.Error(w, "Error reading request body", http.StatusInternalServerError)
-			return
-		}
 
 		switch r.Header.Get("Accept") {
 		case "text/event-stream":
-			event := fmt.Sprintf("saved-%s", address)
+
+			var readyChan = make(chan struct{})
+
+			iterateId := sebakcommon.GetUniqueIDFromUUID()
+			go func() {
+				<-readyChan
+				if ba, err = GetBlockAccount(storage, address); err != nil {
+					http.Error(w, "Error reading request body", http.StatusInternalServerError)
+					return
+				}
+				observer.BlockAccountObserver.Trigger(fmt.Sprintf("iterate-%s", iterateId), ba)
+			}()
+
+			event := fmt.Sprintf("iterate-%s", iterateId)
+			event += " " + fmt.Sprintf("address-%s", address)
 			callBackFunc := func(args ...interface{}) (account []byte, err error) {
-				ba := args[0].(*BlockAccount)
+				//typedEvent := args[0].(string)
+				ba := args[1].(*BlockAccount)
 				if account, err = ba.Serialize(); err != nil {
 					return []byte{}, sebakerror.ErrorBlockAccountDoesNotExists
 				}
 				return account, nil
 			}
-			streaming(observer.BlockAccountObserver, w, event, callBackFunc, s)
+			streaming(observer.BlockAccountObserver, w, event, callBackFunc, readyChan)
+		default:
+			if ba, err = GetBlockAccount(storage, address); err != nil {
+				http.Error(w, "Error reading request body", http.StatusInternalServerError)
+				return
+			}
+
+			var s []byte
+			if s, err = ba.Serialize(); err != nil {
+				http.Error(w, "Error reading request body", http.StatusInternalServerError)
+				return
+			}
+			if _, err = w.Write(s); err != nil {
+				http.Error(w, "Error reading request body", http.StatusInternalServerError)
+				return
+			}
+		}
+	}
+}
+
+const GetAccountTransactionsHandlerPattern = "/account/{address}/transactions"
+
+func GetAccountTransactionsHandler(storage *sebakstorage.LevelDBBackend) http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		vars := mux.Vars(r)
+		address := vars["address"]
+
+		var err error
+		var s []byte
+
+		switch r.Header.Get("Accept") {
+		case "text/event-stream":
+			var readyChan = make(chan struct{})
+			iterateId := sebakcommon.GetUniqueIDFromUUID()
+			go func() {
+				<-readyChan
+				iterFunc, closeFunc := GetBlockTransactionsByAccount(storage, address, false)
```

Comment:
To get the latest `BlockTransaction` of account, the last argument must be `false`, right?

Reply:
to get latest element, the last argument should be `true`.
It is not easy to explain in text. I will explain about this in person. ^^